kong/plugins/rate-limiting/schema.lua

```diff
@@ -59,6 +61,10 @@ return {
       end
     end
 
+    if plugin_t.limit_by == "key" and not plugin_t.key_name then
+      return false, Errors.schema "You need to specify a key name"
+    end
```

Comment:
These schema updates will need to be ported to the new format _after_ https://github.com/Kong/kong/pull/3778 is merged. It converts our plugins schemas to the new schema library we have developed for Services and Routes, and is becoming the new schema library for Kong 1.0.